### PR TITLE
Patch for ISO and Map display

### PR DESCRIPTION
**Pull request by johannrichard**
Implements #43 and adds a (Static) map display on the page. Hope style and naming convention fit you.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,9 @@ Country and city lookup:
 $ http ifconfig.co/country
 Elbonia
 
+$ http ifconfig.co/iso
```
Comment (mpolden) [inline README.md]: I don't think "iso" is self-explanatory. Can you make the path `country-iso`?

File: README.md [hunk 2]
```diff
@@ -43,6 +46,7 @@ $ http --json ifconfig.co
 {
   "city": "Bornyasherk",
   "country": "Elbonia",
+  "isoCountry": "EB",
```
Comment (mpolden) [inline README.md]: Field should follow same naming convention as other fields. Can you make it `country_iso`?

File: api/api.go [hunk 3]
```diff
@@ -84,6 +85,10 @@ func (a *API) newResponse(r *http.Request) (Response, error) {
 	if err != nil {
 		a.log.Debug(err)
 	}
+	iso, err := a.oracle.LookupCountryISO(ip)
```
Comment (mpolden) [inline api/api.go]: `iso` -> `countryISO`.

File: api/oracle.go [hunk 4]
```diff
@@ -127,6 +137,21 @@ func lookupCountry(db *geoip2.Reader, ip net.IP) (string, error) {
 	return "Unknown", fmt.Errorf("could not determine country for IP: %s", ip)
 }
 
+func lookupCountryISO(db *geoip2.Reader, ip net.IP) (string, error) {
+	record, err := db.City(ip)
+	if err != nil {
+		return "", err
+	}
+
```
Comment (mpolden) [inline api/oracle.go]: Please remove blank line.

File: index.html [hunk 5]
```diff
@@ -63,6 +63,10 @@ <h2>Country lookup:</h2>
         <pre>
 $ http {{ .Host }}/country
 {{ .Country }}</pre>
+        <h2>Country ISO lookup:</h2>
```
Comment (mpolden) [inline index.html]: Can you make the text "Short country code (ISO 3166-1 alpha-2) lookup"?

File: index.html [hunk 6]
```diff
@@ -98,6 +103,13 @@ <h2>Testing port connectivity:</h2>
     "port": 8080,
     "reachable": false
 }</pre>
+{{ end }}
+      </div>
+{{ if .IsLookupCityEnabled }}
```
Comment (mpolden) [inline index.html]: Should check that both city and country are not "Unknown".

File: index.html [hunk 7]
```diff
@@ -98,6 +103,13 @@ <h2>Testing port connectivity:</h2>
...
+      <div class="pure-u-1 pure-u-md-1-1">
+        <h2>MAP</h2>
```
Comment (mpolden) [inline index.html]: "MAP" -> "Map".

File: index.html (hunk 6 above)
Comment (johannrichard) [reply index.html]: Not sure the way I implemented it is the "right" way in `Go` terms. I decided to check both for having the Country/City Lookup enabled _and_ to check whether any of the two values returns `"Unknown"`. I didn't find another way to properly chain the checks, hope that's fine.

**Comment by mpolden** [comment]
Thanks!

**Pull request merged**